addons/account/static/src/js/account_report_widgets.js

```diff
@@ -1,604 +1,648 @@
-                    var report_name = window.$("div.page").attr("class").split(/\s+/)[2];
-                    var context_id = window.$("div.page").attr("class").split(/\s+/)[3];
-                    var model = new openerp.Model('account.report.context.common');
-                    model.call('get_context_name_by_report_name', [report_name]).then(function (result) {
-                        var contextModel = new openerp.Model(result);
-                        contextModel.query(['footnotes_manager_id'])
-                        .filter([['id', '=', context_id]]).first().then(function (context) {
-                            var managerModel = new openerp.Model('account.report.footnotes.manager');
-                            managerModel.call('remove_footnote', [[parseInt(context.footnotes_manager_id[0])], parseInt(num)]);
-                        });
-                    });
+                    var diff = dtTo.getTime() - dtFrom.getTime();
+                    dtTo = dtFrom;
+                    dtTo.setDate(dtFrom.getDate() - 1);
+                    dtFrom = new Date(dtTo.getTime() - diff);
+                }                        
+            }
+            else {
+                dtTo.setFullYear(dtTo.getFullYear() - 1);
+                if (!no_date_range) {
+                    dtFrom.setFullYear(dtFrom.getFullYear() - 1);
                 }
-            },
-            fold: function(e) {
-                e.stopPropagation();
-                e.preventDefault();
-                var report_name = window.$("div.page").attr("class").split(/\s+/)[2];
-                var context_id = window.$("div.page").attr("class").split(/\s+/)[3];
+            }
+            if (!no_date_range) {
+                this.$("input[name='date_from_cmp']").val(dtFrom.toISOString().substr(0, 10)); 
+            }
+            this.$("input[name='date_to_cmp']").val(dtTo.toISOString().substr(0, 10)); 
+
+        }
+    },
+    footnoteFromDropdown: function(e) {
+        e.stopPropagation();
+        e.preventDefault();
+        self = this;
+        var report_name = $(e.target).parents("div.page").attr("data-report-name");
+        var context_id = $(e.target).parents("div.page").attr("data-context");
+        var model = new openerp.Model('account.report.context.common');
+        return model.call('get_context_name_by_report_name', [report_name]).then(function (result) {
+            self.curFootNoteTarget = $(e.target).parents("div.dropdown").find("span.account_id");
+            var contextModel = new openerp.Model(result);
+            var type = $(e.target).parents('tr').data('type');
+            var target_id = $(e.target).parents('tr').data('id');
+            var column = $(e.target).parents('td').index();
+            $("#footnoteModal #type").val(type);
+            $("#footnoteModal #target_id").val(target_id);
+            $("#footnoteModal #column").val(column);
+            $('#footnoteModal').on('hidden.bs.modal', function (e) {
```

Comment:
this is more a general remark on good practice and performances:
when you perform multiple jquery selector request that are identical, it is better to "cache" them inside a variable. Let me explain, the DOM is basically a huge String and performing a jquery selector request like $(e.target).parents('tr') will have to goes over every element of the DOM to find the node you're looking for, depending on the size of your DOM it can be fast or not, therefore if you have to perform multiple operation with that node (here it is only reading the type and the id so it's not big deal but sometimes you have more operation to do), it is better to cache the result of the jquery selector inside a variable. By convention when you store a jquery result in a variable, prefix the variable name with $, that way other developers know easily that they deal with.

example: var $this = $(this);
var $variable = $(e.target).parents('tr');

On the same kind of idea, I can see in this particular case that you have 5 times next to each other a call to $('#footnoteModal'), you could cache the result of $('#footnoteModal') inside a variable and then perform query on that variable which will be way faster because in your case we have to look in all the DOM for footnoteModal that contains the id type, then another request that looks over all the DOM again for footnoteModal that contains target_id now, and so on... If you just look for footnoteModal and store it inside a variable, all request after that only have to search through a small portions of text and not all the DOM (I hope it is clear, if not don't hesitate to ask).